### ISSUE TITLE
MAUI 7.0 iOS Release build won't install on iOS 16

### ISSUE THREAD

**Issue by jdunaway** [dotnet/maui#11385]
### Description

I'm using Azure DevOps to create an ad-hoc build for iOS for a .net 7 MAUI application (recently updated from .net 6 which had the same issue):

`dotnet publish -f net7.0-ios -c Release -r ios-arm64 /p:ArchiveOnBuild=true `

The .ipa file produced by the build works on iOS 15 devices but when I download it on an iOS 16 device the icon is blank and when I tap it to open it says the application cannot be installed at this time. Has anyone successfully published a MAUI release build for iOS 16?

### Steps to Reproduce

1. publish ad-hoc release build of .net maui application `dotnet publish -f net7.0-ios -c Release -r ios-arm64 /p:ArchiveOnBuild=true `
2. Download it on an iOS 16 device.  
3. For me, the icon is blank and when I click the application I see a message saying "Unable to Install  Please try again later."

### Link to public reproduction project repository

(https://github.com/jdunaway/MauiAppReproiOS16ReleaseBuildIssue)

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iOS 16

### Did you find any workaround?

No.

**Comment by tj-devel709** [comment]
Hi @jdunaway, thank you for your feedback!

Starting in iOS 16, you will need to enable "Developer Mode" on the iphone in order to install .ipa files.
https://developer.apple.com/documentation/xcode/enabling-developer-mode-on-a-device

> "The feature doesn’t affect ordinary installation techniques like buying apps from the App Store, or participating in a TestFlight team. Instead, Developer Mode focuses on scenarios like performing a Build and Run in Xcode, or installing an .ipa file with [Apple Configurator](https://support.apple.com/apple-configurator)."

Please try this and let us know if this fixes your issue. Thank you!

**Comment by jdunaway** [comment]
Hi @tj-devel709.  Thanks for the reply.  This issue is only happening on the ad-hoc release build, which shouldn't require Developer mode.  I do have developer mode enabled on the device though since it was required in order to run the debug build on the device.  The debug build is working, but the release build will not install or open on the device.

**Comment by PureWeen** [comment]
@rolfbjarne thoughts?

@jdunaway if you just use a new MAUI app does that work?

**Comment by jdunaway** [comment]
@PureWeen Haven't tried that yet but will try when I get a chance and reply back.  If it does still happen will also create a repo with the issue.  Can't create a repo of the current solution since it has proprietary code in it.

**Comment by samirgcofficial** [comment]
sudo dotnet publish -f net7.0-ios -c Release -r ios-arm64 /p:ArchiveOnBuild=true 
This  code  will pick up developer certificate instead of Release certificate under release configuration, I have added both the certificate inside Debug and Release build.  (.net 7 VS Mac 2022 ) any one facing same issue ?

**Comment by rolfbjarne** [comment]
> @rolfbjarne thoughts?

This looks weird, but the first step would be to figure out whether this is app-specific (does it happen with a new MAUI app?) or if it's something specific to one project (in which case we'll need a repro project).

**Comment by jdunaway** [comment]
I was able to reproduce this issue with a file -> new maui project.  Here is the repo: https://github.com/jdunaway/MauiAppReproiOS16ReleaseBuildIssue  Please see the azure-pipelines.yml file for how I'm building this.  I've also tried building this locally on the windows and mac side but I'm currently getting a Link.semaphore error.  So you may want to try building this locally too.  You will need to update the certificate to an adhoc iOS distribution certificate and update the provisioning profile to your own provisioning profile in order to reproduce.  Also note the project settings I have set for the release build if you think that may be leading to the issue.  These are the same as what I have in our production project that we are trying to release but can't release until the iOS 16 install issue is resolved.  I get the same results with the reproduction repo: the ipa produced will install on iOS 15 devices but not iOS 16 devices.  I'm able to debug on both iOS 15 and iOS 16 devices but the release build won't install on iOS 16 devices.

**Comment by jdunaway** [comment]
Please reopen this issue since the same issue occurs with a file -> new maui project and I have provided the reproduction repo in the previous comment.

**Comment by jdunaway** [comment]
@rolfbjarne @PureWeen @jsuarezruiz Please remove the needs-info label since the info has been provided.

**Comment by jdunaway** [comment]
We have resolved this issue based on this post in the apple developer forum: https://developer.apple.com/forums/thread/714600  The issue was with the bundle-identifier in our manifest.plist file.  At one point we added Maui to the end of the bundle-identifier but removed it later and didn't update the manifest.plist file.  It looks like this isn't an issue for iOS 15 but iOS 16 is more strict and won't allow you to install the application if the bundle-identifier isn't an exact match.  Removing Maui from the bundle-identifier in the manifest.plist resolved this issue.
